### ISSUE TITLE
Anonymous functions create closures (or something like them)

### ISSUE THREAD

**Issue by sjbarag** [sjbarag/brs#61]
Per [BrightScript's documentation](https://sdkdocs.roku.com/display/sdkdoc/Program+Statements#ProgramStatements-AnonymousFunctions) (emphasis mine):

> A function is anonymous if it does not have a name. **Note that Anonymous Functions do not currently create closures.**

Consider the following BrightScript file:

```brightscript
function Main()
    a = "before"

    showA = sub()
        print a
    end sub

    showA()

    a = "after"
    showA()
end function

Main()
```

The reference implementation prints

```
<UNINITIALIZED>
<UNINITIALIZED>
```

which indicates that the value of `a` wasn't captured by the creation of the anonymous function `showA`.  The current `brs` implementation _does_ capture the value of `a`:

```
before
after
```

While closures are certainly useful, their presence in `brs` will lead to this project supporting an incompatible superset of the reference implementation.  Programs that execute as-expected in `brs` will behave unexpectedly in the reference implementation due to its different scoping rules, and vice-versa.

**Comment by sjbarag** [comment]
Added milestone "BRS 0.9" because this *must* be fixed before the next public release.

**Comment by sjbarag** [comment]
Fixed in #66
